loader/linux/icd_linux.c

```diff
@@ -197,10 +197,19 @@ static inline void khrIcdOsDirEnumerate(const char *path, const char *env,
                 }
                 free(dirElems);
             }
-        } else
+        }
+        else
+        {
             // use system provided ordering
             for (dirEntry = readdir(dir); dirEntry; dirEntry = readdir(dir) )
-                khrIcdOsDirEntryValidateAndAdd(dirEntry->d_name, path, extension, addFunc);
+            {
+                // only check entries different from '.' and '..'
+                if (strcmp(dirEntry->d_name, ".") && strcmp(dirEntry->d_name, ".."))
+                {
+                    khrIcdOsDirEntryValidateAndAdd(dirEntry->d_name, path, extension, addFunc);
+                }
+            }
+        }
```

Comment:
Not against this change, but it should be documented that the loader will refuse to load something if the current or parent directory is specified relatively.
Conceptually I don't see the difference between:
`OCL_ICD_VENDORS=../vendors`
and
`OCL_ICD_VENDORS=.`
or maybe I don't understand the reason for this change.

Reply:
Hi, sorry for the late answer. If I remember correctly this was changed because of an error on Mac when manually specifying the path to the vendors' directory. The change is only for the loop that checks the `.icd` files in the directory specified (so files `.` and `..` are not checked), so I don't think this adds a requirement for the folder path to be absolute.

However, I did try to reproduce it but I was not able to do so. So if the change is not wanted the commit can be dropped I guess.

Reply:
I think it is better to remove it. Functional changes should go in their own PR.

Reply:
Rebased and removed this change.